apps/src/templates/codeDocs/ProgrammingExpressionOverview.jsx

```diff
@@ -22,26 +28,61 @@ export default function ProgrammingExpressionOverview({programmingExpression}) {
         VIDEO_HEIGHT
       );
     }
-  });
-  // Spritelab passes down its color in HSL format, whereas other labs use hex color
-  const color =
-    programmingExpression.color &&
-    (typeof programmingExpression.color === 'string'
-      ? programmingExpression.color
-      : `hsl(${programmingExpression.color[0]},${programmingExpression
-          .color[1] * 100}%, ${programmingExpression.color[2] * 100}%)`);
+    if (titleRef.current && programmingExpression.blockName) {
+      convertXmlToBlockly(titleRef.current);
+      const blocksDom = parseElement(
+        `<block type='${programmingExpression.blockName}' />`
+      );
+      const blockSpace = Blockly.BlockSpace.createReadOnlyBlockSpace(
+        titleRef.current,
+        blocksDom,
+        {
+          noScrolling: true,
+          inline: false
+        }
+      );
+      shrinkBlockSpaceContainer(blockSpace, true);
+    }
+  }, [programmingExpression]);
+
+  const getColor = () => {
+    if (!programmingExpression.color) {
```

Comment:
super style nit: we reference `programmingExpression.color` a few times in this function, so i'd pull it into its own variable for readability